### DISCUSSION TITLE
Deploy Hooks & Ignored Build Step

### DISCUSSION THREAD

**Discussion by yonbergman** [vercel/community#3388] [Help] [answered]
### Summary

Is there a way to effectively use Deploy Hooks and Ignored Build Step together? We have a monorepo where Ignored Build Step is configured to deploy changes only in the relevant package. However, we also want to trigger rebuilds for specific content changes using Deploy Hooks. The problem is that sometimes the deploy is canceled because the Ignored Build Step detects no relevant file changes in the main branch.

### Steps to Reproduce

Have a project set up with a Ignored Build Step like `git diff HEAD^ HEAD --quiet .` and run a deploy hook after committing a change into the git repo outside of the project folder.

**Comment by webspaceadam** [comment]
Did you find a solution for that? Every solution i found is veeeery hacky 😅

**Comment by amyegan** [comment, marked as answer]
Thanks for reaching out. If you're using Turborepo, there's a relatively new preset option for [Ignored Build Step](https://vercel.com/docs/projects/overview#ignored-build-step) that can help with this. Or you can write your own more customized script to suit your project's needs.

https://vercel.com/guides/how-do-i-use-the-ignored-build-step-field-on-vercel
